lib/active_record/connection_adapters/sqlserver/database_statements.rb

```diff
@@ -333,11 +408,39 @@ def handle_more_results?(handle)
 
         def handle_to_names_and_values(handle, options = {})
           case @connection_options[:mode]
+          when :jdbc
+            handle_to_names_and_values_jdbc(handle, options)
           when :dblib
             handle_to_names_and_values_dblib(handle, options)
           end
         end
 
+        def handle_to_names_and_values_jdbc(handle, options = {})
+          query_options = {}.tap do |qo|
+            qo[:as] = (options[:ar_result] || options[:fetch] == :rows) ? :array : :hash
+            qo[:database_timezone] = ActiveRecord::Base.default_timezone || :utc
+          end
+
+          results =
+            begin
+              handle.all(query_options)
+            rescue => e
```

Comment:
Same trap on rescue, would you mind letting me know why no results returns an error. Can that be avoided? I'd really like to know why the JDBC connection wold do that.